config/clusters/hhmi/spyglass.values.yaml

```diff
@@ -0,0 +1,84 @@
+nfs:
+  enabled: true
+  dirsizeReporter:
+    enabled: true
+  pv:
+    mountOptions:
+      - soft
+      - noatime
+    serverIP: 10.55.112.74
+    baseShareName: /homes/
+jupyterhub:
+  ingress:
```

Comment:
I've also documented now that we should enable the hook prepuller and disable the configurator. https://github.com/2i2c-org/infrastructure/pull/3653

Reply:
I get a validation error and cannot deploy if I don't define the rest, so there is a bug in our schema:

```
1 / 1: Validating non-encrypted hub values files for spyglass...
Error: values don't meet the specifications of the schema(s) in the following chart(s):
basehub:
- nfs.pv: Must validate at least one schema (anyOf)
- nfs.pv: serverIP is required
- nfs.pv: baseShareName is required
```

Reply:
See [`bcb736c` (#3646)](https://github.com/2i2c-org/infrastructure/pull/3646/commits/bcb736cabc7c993d5dddb029aac2b45a056d48c0) for change

Reply:
This is the related schema section https://github.com/2i2c-org/infrastructure/blob/cfb4e641b7993922c640f523bae87010d844b309/helm-charts/basehub/values.schema.yaml#L189-L201

Reply:
This config fixed it:

```
nfs:
  enabled: false
  pv:
    enabled: false
```

But I am surprised that the top level `nfs.enabled = false` didn't automatically turn off the nested `pv` section

Reply:
Hmmm, so there is an unconditional requirement to have `nfs.pv` declared:

https://github.com/2i2c-org/infrastructure/blob/cfb4e641b7993922c640f523bae87010d844b309/helm-charts/basehub/values.schema.yaml#L156-L163

basehub's default values declares `nfs.pv`, and has `nfs.pv.enabled=True`:

https://github.com/2i2c-org/infrastructure/blob/cfb4e641b7993922c640f523bae87010d844b309/helm-charts/basehub/values.yaml#L43-L48

The schema section for `nfs.pv` is conditional, where more fields are required when `nfs.pv.enabled=True`.

https://github.com/2i2c-org/infrastructure/blob/cfb4e641b7993922c640f523bae87010d844b309/helm-charts/basehub/values.schema.yaml#L189-L204

---

I think the improvement is to not default to `nfs.pv.enabled=true` since other fields are required to be set for that, but instead default nfs.pv.enabled=false` and set it to true alongside other required fields.

Opening an issue about this.

Reply:
Tracked in https://github.com/2i2c-org/infrastructure/issues/3654